clang/test/CIR/Lowering/loop.cir

```diff
@@ -95,3 +93,65 @@ module {
 //  LLVM-NEXT: 15:
 //  LLVM-NEXT:   ret void
 //  LLVM-NEXT: }
+
+  // Test while cir.loop operation lowering.
+  cir.func @testWhile(%arg0: !s32i) {
+    %0 = cir.alloca !s32i, cir.ptr <!s32i>, ["i", init] {alignment = 4 : i64}
+    cir.store %arg0, %0 : !s32i, cir.ptr <!s32i>
+    cir.scope {
+      cir.loop while(cond : {
```

Comment:
I would prefer that too.

I don't think it would change much in functionality, but it would improve the code organization, I think. As far as I see, the loop interface would be all we need to avoid duplication of pass logic in the operations.